unstructured_ingest/v2/processes/connectors/notion/connector.py

```diff
@@ -0,0 +1,322 @@
+from dataclasses import dataclass
+from time import time
+from typing import Any, Generator, Optional
+
+from pydantic import UUID4, Field, Secret
+
+from unstructured_ingest.error import SourceConnectionError
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+NOTION_API_VERSION = "2022-06-28"
+CONNECTOR_TYPE = "notion"
+
+
+class NotionAccessConfig(AccessConfig):
+    notion_api_key: str = Field(description="Notion API key")
+
+
+class NotionConnectionConfig(ConnectionConfig):
+    access_config: Secret[NotionAccessConfig]
+
+
+class NotionIndexerConfig(IndexerConfig):
+    page_ids: Optional[list[str]] = Field(
+        default=None, description="List of Notion page IDs to process"
+    )
+
+    database_ids: Optional[list[str]] = Field(
+        default=None, description="List of Notion database IDs to process"
+    )
+    recursive: bool = Field(
+        default=False, description="Recursively process child pages and databases"
+    )
+
+    def __post_init__(self):
+        if self.page_ids:
+            self.page_ids: list[UUID4] = [UUID4(p.strip()) for p in self.page_ids]
+
+        if self.database_ids:
+            self.database_ids: list[UUID4] = [UUID4(p.strip()) for p in self.database_ids]
+
+
+@dataclass
+class NotionIndexer(Indexer):
+    connection_config: NotionConnectionConfig
+    index_config: NotionIndexerConfig
+
+    @requires_dependencies(["notion_client"], extras="notion")
+    def get_client(self):
```

Comment:
But now every place that uses this method needs to call the async methods and have those methods also wrapped in `async` and `await` where appropriate.